### PR TITLE
Add node module resolver to fix missing bsb bin in hoisted projects

### PR DESCRIPTION
**Pull request by dmisdm**
A bit confused as to how my reformatter produced something different to what was there...
If its a problem ill fix it.

### CODE REVIEW

**Comment by dmisdm** [comment]
I also wasn't sure how to tackle the case when bs-platform can't be found. At the moment it causes an unhandled `failwith("module bs-platform not found")` (look at BuildSystem.bsPlatformDir).
Trying to be pragmatic so that consumers don't have to handle a Result, but maybe thats best?

**Comment by dmisdm** [comment]
Realising this needs some more work so closing for now

**Comment by jaredly** [comment]
I only use refmt occasionally, so that's why it produced different formatting :)

File: src/analyze/BuildSystem.re [hunk 1]
```diff
@@ -10,25 +9,48 @@ type t =
...
+      ++ "\n\n"
+      ++ String.concat("\n", out @ err),
+    )
+  };
 
 open Infix;
+
+/* Not sure if raising an exception is the best option... */
+let bsPlatformDir =
+  switch (ModuleResolution.resolveNodeModulePath("bs-platform")) {
```
Comment (jaredly) [inline src/analyze/BuildSystem.re]: this needs to be resolved in relation to `rootPath` -- don't cache it like this

**Comment by dmisdm** [comment]
apologies! hope its ok now.
would you be keen on making a commit that just reformats the whole project before other changes are made?

**Comment by jaredly** [comment]
Fantastic thanks! I'm currently in process of rewriting the internal code analysis engine, so I don't want to have too many changes on the main branch. After that merge, I'll be open to refmting everything.

**Comment by dmisdm** [comment]
Awesome

**Pull request merged**